### DISCUSSION TITLE
How does it compare to LuaLatex ?

### DISCUSSION THREAD

**Discussion by FabriceSalvaire** [tectonic-typesetting/tectonic#963] [Q&A]
As far I know, the most advanced rework of the original TeX Web code was [LuaLatex](https://www.luatex.org).
**So my question is** : How does it compare to LuaLatex ?

- A Git repository is nice to have.
- An active community ? LuaLaTex home page seems a bit unmaintained...
- LuaLaTex implements a lot of hooks.
- LuaLaTex implements Lua coding in TeX which is a powerful feature and might be a better solution than generating TeX code by another program.

**Comment by Neved4** [comment]
Hey @FabriceSalvaire I really like this idea! 🚀

Having a comparison table between `tectonic` and the alternatives (e.g. engines: pdfLaTeX, LuaLaTeX, XeTeX; tools: latexmk, MikTeX, arara, llmk...) that outlines `tectonic` **strengths** would greatly communicate **the goals** of the project 💯

iirc, LuaTeX has a [git repo](https://github.com/TeX-Live/luatex) too.

Naming a few points where `tectonic` really shines:
- **Performance**: LuaTeX is slower than other engines (main competitors: `pdfLaTeX`, `XeTeX`)
- **UX**: really nice CLI, and a cargo-style document model (main competitors: `latexmk`, `llmk`)
- **On-demand download**: packages are pulled when needed for compilation, no need to carry ≈4GB everywhere (main competitors: `MikTeX`)
- **Community**: `tectonic`'s community it's vibrant. It has pulled so many people from the TeX Live world. People that wanted to use it for their projects, and reported what wasn't working at the time or could've been improved (`biber`, `minted`, `tikz`...). Even when `tectonic` wasn't entirely ready or implied tradeoffs at different stages, it received a lot of attention, cherish and positive involvement.
- **Reputation**: the project has a history of fixing things to _serve_ the users needs.

**Comment by pkgw** [comment]
Thanks for the enthusiasm, @mnrvwl! Regarding the git repo: the LuaTeX one is labeled as a Git mirror of the primary SVN repository. In my opinion, having the *primary* repository be Git-based (and hosted on a social site like GitHub, GitLab, etc.) is an important differentiator.

I'm going to convert this to a Discussion since it's not an "issue" to be resolved, although (as ever) it indicates a lack of documentation that would be good to resolve.
